Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -283,10 +283,64 @@ CFTypeID CGPathGetTypeID() {
     return __CGPath::GetTypeID();
 }
 
+// A helper for determining the number of points per path element type
+static inline int CGPathGetExpectedPointCountForType(CGPathElementType type) {
+    return type == kCGPathElementMoveToPoint || type == kCGPathElementAddLineToPoint ?
+               1 :
+               type == kCGPathElementAddQuadCurveToPoint ? 2 : type == kCGPathElementAddCurveToPoint ? 3 : 0;
+}
+
+// Create a mimic of CGPathElement that holds the points in a vector with a convenient copy constructor.
+struct _CGPathElementVector {
+    CGPathElementType _type;
+    std::vector<CGPoint> _points;
+    _CGPathElementVector(const CGPathElement el)
+        : _type(el.type), _points(el.points, el.points + CGPathGetExpectedPointCountForType(el.type)) {
+    }
+};
+
+// A struct to pass to the equality matching CGPathApply since only a single void* may be passed.
+struct CGPathElementMatch {
+    std::vector<_CGPathElementVector> elements;
+    bool equal = true;
+    int positionToMatch = 0;
+};
+
+// A function to pass to CGPathApply to determine whether two paths are equal.
+static void _CGPathApplyCheckEquality(void* pathElements, const CGPathElement* element1) {
+    if (!((CGPathElementMatch*)pathElements)->equal) {
```

Comment:
I'm not sure what you mean here, could you elaborate?

---
In reply to: [97700178](https://github.com/Microsoft/WinObjC/pull/1813#discussion_r97700178) [](ancestors = 97700178)